lib/repl.js

```diff
@@ -851,14 +851,15 @@ function complete(line, callback) {
           // Exclude versioned names that 'npm' installs.
           continue;
         }
-        if (exts.indexOf(ext) !== -1) {
+        abs = path.resolve(dir, name);
+        isDirectory = fs.statSync(abs).isDirectory();
```

Comment:
It would be a race condition that is hart to trigger but as there was a guard for this before I would say it should be kept as it was.